lib/src/klio/transforms/decorators.py

```diff
@@ -184,170 +188,218 @@ def __from_klio_message_generator(self, kmsg, payload, orig_item):
         # executes the next `yield`
         return
 
+    else:
+        metrics.success.inc()
+
 
 # meant for DoFn.process generator methods; very similar to the
 # '__serialize_klio_message' function, but needs to be its own function
 # since it's a generator (can't mix the two - if a function has a `yield`
 # statement, it automatically becomes a generator)
 def __serialize_klio_message_generator(
-    self, meth, incoming_item, *args, **kwargs
+    metrics, self, meth, incoming_item, *args, **kwargs
```

Comment:
Since this is not an actual method, `self` is actually just a reference to the instance being passed around (it'd prob be a good idea to rename as `inst` or `instance` rather than `self`). I put `metrics` first because it's the only argument that is used only for these private helper functions, and is _not_ used in the user function (that the private helper functions actually invoke). Basically, everything after `metrics` is used or passed to the user function/method; I thought it'd be more confusing to inter-mingle those args.